vip-api/src/main/java/fr/insalyon/creatis/vip/api/SpringSecurityConfig.java

```diff
@@ -72,41 +91,64 @@
     @Autowired
     private Environment env;
 
+    @Autowired
+    private ApiBusiness apiBusiness;
+
+
+    @Override
+    protected SessionAuthenticationStrategy sessionAuthenticationStrategy() {
+        return new RegisterSessionAuthenticationStrategy(new SessionRegistryImpl());
+    }
+
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) {
+        KeycloakAuthenticationProvider keycloakAuthenticationProvider = keycloakAuthenticationProvider();
+        keycloakAuthenticationProvider.setGrantedAuthoritiesMapper(grantedAuthoritiesMapper());
+        auth.authenticationProvider(keycloakAuthenticationProvider);
+    }
+
     @Override
     protected void configure(HttpSecurity http) throws Exception {
+        super.configure(http);
         http
-            .authorizeRequests()
+                .authorizeRequests()
                 .antMatchers("/rest/platform").permitAll()
                 .antMatchers("/rest/authenticate").permitAll()
+                .antMatchers("/rest/register").permitAll() //signup a user to VIP
+                .antMatchers("/rest/simulate-refresh").authenticated()
                 .antMatchers("/rest/statistics/**").hasAnyRole("ADVANCED", "ADMINISTRATOR")
                 .antMatchers("/rest/**").authenticated()
                 .anyRequest().permitAll()
-            .and()
-            .apply(new ApikeyAuthentificationConfigurer<>(
-                    env.getRequiredProperty(CarminProperties.APIKEY_HEADER_NAME),
-                    apikeyAuthenticationEntryPoint))
-            .and()
-            .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
-            .and()
-            .cors().and()
-            .headers().frameOptions().sameOrigin().and()
-            .csrf().disable();
+                //trying to test the apikeyauthenticator
+                .and()
+                .apply(new ApikeyAuthentificationConfigurer<>(
+                        env.getRequiredProperty(CarminProperties.APIKEY_HEADER_NAME),
+                        apikeyAuthenticationEntryPoint))
+                .and()
+                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+                .and()
+                .cors().and()
+                .headers().frameOptions().sameOrigin().and()
+                .csrf().disable();
     }
 
+
     @Bean
     public Supplier<User> currentUserProvider() {
-        return () -> {
```

Comment:
Does this still work when the user is authenticated with an API key ?

Reply:
the api is not secured by the api key all requests needs keycloak JWT

Reply:
But for the current VIP service we still need to make it work with an API key as today.
We need a property (or a spring profile to activate one of the 2 modes (keycloak / api key)